File: gems/decomplex/examples/csharp/locality-drag.cs
class Example {
  static void run(User user, Cart cart, Logger logger) {
    var receipt_id = user.id;

    var total = cart.total;
    if (total > 100) {
      if (cart.discountable()) {
        var discount = 10;
      }
    }
    if (cart.taxable()) {
      if (cart.region) {
        var tax = total * 2;
      }
    }
    if (logger.enabled()) {
      if (logger.debug()) {
        logger.info(total);
      }
    }
    if (cart.valid()) {
      if (cart.ready()) {
        var status = 1;
      }
    }

    emit(receipt_id);
  }
}

File: gems/decomplex/examples/go/locality-drag.go
package main
func run(user User, cart Cart, logger Logger) {
  receipt_id := user.id

  total := cart.total
  if total > 100 {
    if cart.discountable() {
      discount := 10
      _ = discount
    }
  }
  if cart.taxable() {
    if cart.region {
      tax := total * 2
      _ = tax
    }
  }
  if logger.enabled() {
    if logger.debug() {
      logger.info(total)
    }
  }
  if cart.valid() {
    if cart.ready() {
      status := 1
      _ = status
    }
  }

  emit(receipt_id)
}
